xla/service/gpu/gpu_fusible.cc

```diff
@@ -330,9 +330,41 @@ bool IsLoopFusibleAsProducer(const HloInstruction& instr) {
           (instr.opcode() == HloOpcode::kIota ||
            instr.opcode() == HloOpcode::kConstant ||
            // Non-variadic elemental reductions can be fused as producers.
-           (instr.opcode() == HloOpcode::kReduce &&
-            !IsReductionFromOrToContiguousDimensions(instr) &&
-            !instr.shape().IsTuple())));
+           (instr.opcode() == HloOpcode::kReduce && !instr.shape().IsTuple())));
+}
+
+static bool AllSatisfy(const HloInstruction& instr,
+                       const HloPredicate& predicate) {
+  if (instr.opcode() != HloOpcode::kFusion) {
+    return predicate(&instr);
+  }
+
+  return absl::c_all_of(
+      instr.fused_instructions(), [&](const HloInstruction* i) {
+        return i->opcode() == HloOpcode::kParameter || predicate(i);
+      });
+}
+
+bool IsReduceIntermediate(const HloInstruction* instr) {
+  if (instr->operand_count() > 1 || instr->user_count() > 1) {
+    return false;
+  }
+
+  // Only support elementwise ops that don't introduce additional compute.
+  // More benchmarking and better cost model are needed to enable this for
+  // more compute ops.
+  switch (instr->opcode()) {
+    case HloOpcode::kBitcast:
+    case HloOpcode::kBitcastConvert:
+    case HloOpcode::kConvert:
+    case HloOpcode::kCopy:
```

Comment:
Is there a practical use case for such reduce->copy fusion? If so - could you add a test for it?

Reply:
i havent seen a concrete case for reduce-copy, it's just something i thought would be nice to have.. i can remove this in this pr